app/services/dashboard_service.rb

```diff
@@ -0,0 +1,75 @@
+class DashboardService 
```

Comment:
Ideally your service fits better as a stateless object performing a single action. It seems to me that this service acts more like a query object than a service itself, for example:

```ruby
OrderedEventsQuery.new
UpcomingWorkshopsQuery.new
```